pkg/cloud/scope/rosacontrolplane.go

```diff
@@ -53,11 +59,23 @@ func NewROSAControlPlaneScope(params ROSAControlPlaneScopeParams) (*ROSAControlP
 	}
 
 	managedScope := &ROSAControlPlaneScope{
-		Logger:       *params.Logger,
-		Client:       params.Client,
-		Cluster:      params.Cluster,
-		ControlPlane: params.ControlPlane,
-		patchHelper:  nil,
+		Logger:         *params.Logger,
+		Client:         params.Client,
+		Cluster:        params.Cluster,
+		ControlPlane:   params.ControlPlane,
+		patchHelper:    nil,
+		controllerName: params.ControllerName,
+	}
+
+	session, serviceLimiters, err := sessionForClusterWithRegion(params.Client, managedScope, *params.ControlPlane.Spec.Region, params.Endpoints, params.Logger)
+	if err != nil {
+		return nil, errors.Errorf("failed to create aws session: %v", err)
+	}
+
+	stsClient := NewSTSClient(managedScope, managedScope, managedScope, managedScope.ControlPlane)
```

Comment:
we have to set
```go
managedScope.session = session
managedScope.serviceLimiters = serviceLimiters
```
before calling `NewSTSClient()`

Reply:
Good catch, thanks. We could also have every consumer make their own client and not in-line the identity but it seemed easier to do this for now